zap/lib/src/dom/dom.dart

```diff
@@ -1,24 +1,31 @@
 import 'dart:async';
-import 'dart:html';
+import 'dart:js_interop';
+import 'dart:js_interop_unsafe';
 
-import 'package:js/js_util.dart' as js;
+import 'package:web/web.dart';
 
 extension ZapText on Text {
+  @JS('__text')
+  external ExternalDartReference<String?> __text;
+
   set zapText(String value) {
-    if (wholeText != value) data = value;
+    if (__text.toDartObject != value) {
```

Comment:
This feels like a premature optimization, I think we can `wholeText.toDart != value` in all cases.